internal/attestation/measurements/measurement-generator/generate.go

```diff
@@ -0,0 +1,256 @@
+	)
+}
+
+func byteArrayCompositeLit(hex [32]byte, pos token.Pos) *ast.CompositeLit {
+	var elts []ast.Expr
+	curPos := pos + 16 // 16 = len("[32]byte{") + padding
+	for i, b := range hex {
+		elts = append(elts, &ast.BasicLit{
+			ValuePos: curPos,
+			Kind:     token.INT,
+			Value:    fmt.Sprintf("0x%02x", b),
+		})
+		if (i+1)%8 == 0 {
+			curPos += 11 // line break
+		} else {
+			curPos += 6 // 6 = len("0x00, ")
+		}
+	}
+	return &ast.CompositeLit{
+		Type: &ast.ArrayType{
+			Lbrack: pos,
+			Len:    ast.NewIdent("32"),
+			Elt:    ast.NewIdent("byte"),
+		},
+		Lbrace: pos + 8, // 8 = len("[32]byte")
+		Elts:   elts,
+		Rbrace: pos + 223, // 223 = len("[32]byte{0x00, 0x01, ...}") - 1
+	}
+}
+
+func measurementsEntryKeyValueExpr(pcr uint32, measuremnt measurements.Measurement, pos token.Pos) (*ast.KeyValueExpr, token.Pos) {
+	key := fmt.Sprintf("%d", pcr)
+	keyLen := len(key)
+	colon := pos + token.Pos(keyLen)
+	valuePos := token.Pos(colon + 2)
+	expectedKeyPos := token.Pos(valuePos + 5)
+	expectedColon := token.Pos(expectedKeyPos + 8)
+	expectedValuePos := token.Pos(expectedColon + 2)
+	warnOnlyKeyPos := token.Pos(expectedColon + 1 + byteArrayCompositeLitWidth)
+	warnOnlyColon := token.Pos(warnOnlyKeyPos + 9)
+	warnOnlyValuePos := token.Pos(warnOnlyColon + 2)
+	var rbrace token.Pos
+	if measuremnt.WarnOnly {
+		rbrace = token.Pos(warnOnlyValuePos + 9) // 9 = len("true") + padding
+	} else {
+		rbrace = token.Pos(warnOnlyValuePos + 10) // 10 = len("false") + padding
+	}
+
+	return &ast.KeyValueExpr{
+		Key: &ast.BasicLit{
+			ValuePos: pos,
+			Kind:     token.INT,
+			Value:    key,
+		},
+		Colon: colon,
+		Value: &ast.CompositeLit{
+			// Type:   ast.NewIdent("Measurement"),
+			Lbrace: valuePos,
+			Elts: []ast.Expr{
+				&ast.KeyValueExpr{
+					Key:   &ast.Ident{NamePos: expectedKeyPos, Name: "Expected"},
+					Colon: expectedColon,
+					Value: byteArrayCompositeLit(measuremnt.Expected, expectedValuePos),
+				},
+				&ast.KeyValueExpr{
+					Key:   &ast.Ident{NamePos: warnOnlyKeyPos, Name: "WarnOnly"},
+					Colon: warnOnlyColon,
+					Value: &ast.Ident{NamePos: warnOnlyValuePos, Name: strconv.FormatBool(measuremnt.WarnOnly)},
+				},
+			},
+			Rbrace: rbrace,
+		},
+	}, rbrace + 1
+}
+
+func measurementsCompositeLiteral(measurements measurements.M, pos token.Pos) *ast.CompositeLit {
+	lbrace := pos + 1
+	var elts []ast.Expr
+	pcrs := make([]uint32, 0, len(measurements))
+	for pcr := range measurements {
+		pcrs = append(pcrs, pcr)
+	}
+	sort.Slice(pcrs, func(i, j int) bool { return pcrs[i] < pcrs[j] })
+	entryPos := lbrace + 5
+	for _, pcr := range pcrs {
+		kvExpr, newEntryPos := measurementsEntryKeyValueExpr(pcr, measurements[pcr], entryPos)
+		elts = append(elts, kvExpr)
+		entryPos = newEntryPos + 4 // 4 = padding and newline
+	}
+	return &ast.CompositeLit{
+		Lbrace: lbrace,
+		Type: &ast.Ident{
+			NamePos: pos,
+			Name:    "M",
+		},
+		Elts:   elts,
+		Rbrace: entryPos - 1,
+	}
+}
```

Comment:
Please add a decent amount of documentation to this code, so someone who changed a thing in the target file can update this code in a reasonable time.

Reply:
Also cool stuff by the way...